pypesto/sample/pymc.py

```diff
@@ -2,56 +2,93 @@
 from __future__ import annotations
 
 import logging
-import warnings
-from typing import TYPE_CHECKING, Union
+from typing import Union
 
+import aesara.tensor as at
+import arviz as az
 import numpy as np
+import pymc
 
 from ..history import MemoryHistory
+from ..objective import ObjectiveBase
 from ..problem import Problem
 from ..result import McmcPtResult
 from .sampler import Sampler, SamplerImportError
 
 logger = logging.getLogger(__name__)
 
-if TYPE_CHECKING:
-    try:
-        import aesara.tensor as at
-        import arviz as az
-        import pymc
-
-        from ..objective.aesara import AesaraObjectiveRV
-    except ImportError:
-        raise ImportError(
-            "Using the pymc sampler requires an installation of the "
-            "python packages aesara, arviz and pymc. Please install "
-            "these packages via `pip install aesara arviz pymc`."
-        )
+# implementation based on:
+# https://www.pymc.io/projects/examples/en/latest/case_studies/blackbox_external_likelihood_numpy.html
 
 
-class AesaraDist(pymc.NoDistribution):
-    """PyMC distribution wrapper for AesaraObjectiveRVs."""
+class AesaraObjectiveOp(at.Op):
```

Comment:
seems pretty redundant with https://github.com/ICB-DCM/pyPESTO/blob/488eca74bf5b6abe0a0c60de0215997c6973c0a8/pypesto/objective/aesara.py#L151

Reply:
it certainly is. however, I could not get the pypesto/objective approach to run, kept running into errors, partly induced by recent changes in pymc. this one here is now based on a standard template for blackbox objectives (referenced in the code). :grimacing:

Reply:
Ah, okay. Then let me have another look tomorrow. Would prefer not to duplicate functionality and have the caching implemented in AmiciObjective (which might be the actual problem).